evalml/tests/automl_tests/test_automl.py

```diff
@@ -5214,3 +5239,72 @@ def test_automl_chooses_engine(engine_choice, X_y_binary):
             automl = AutoMLSearch(
                 X_train=X, y_train=y, problem_type="binary", engine=engine_choice
             )
+
+
+@patch("evalml.pipelines.components.RandomForestClassifier.predict_proba")
+@patch("evalml.pipelines.components.RandomForestClassifier.predict")
+@patch("evalml.pipelines.components.ElasticNetClassifier.predict_proba")
+@patch("evalml.pipelines.components.ElasticNetClassifier.predict")
+def test_automl_enesmbler_allowed_component_graphs(
+    mock_en_predict,
+    mock_en_predict_proba,
+    mock_rf_predict,
+    mock_rf_predict_proba,
+    X_y_regression,
+    caplog,
+):
+    """
+    Test that ensures that pipelines that are defined as strings or classes are able to be ensembled.
+    """
+    X, y = X_y_regression
+    mock_en_predict_proba.return_value = np.ones(len(y))
+    mock_rf_predict_proba.return_value = np.ones(len(y))
+    mock_en_predict.return_value = np.ones(len(y))
+    mock_rf_predict.return_value = np.ones(len(y))
+    component_graphs = {
+        "Pipeline1": {
+            "Imputer": ["Imputer", "X", "y"],
+            "Log Transformer": ["Log Transformer", "X", "y"],
+            "RF": ["Random Forest Regressor", "Imputer.x", "Log Transformer.y"],
+        },
+        "Pipeline2": {
+            "Imputer": ["Imputer", "X", "y"],
+            "Log Transformer": ["Log Transformer", "X", "y"],
+            "EN": ["Elastic Net Regressor", "Imputer.x", "Log Transformer.y"],
+        },
+    }
+    automl = AutoMLSearch(
+        X,
+        y,
+        "regression",
+        allowed_component_graphs=component_graphs,
+        ensembling=True,
+        max_batches=4,
+        verbose=True,
+    )
+    automl.search()
+    assert "Stacked Ensemble Regression Pipeline" in caplog.text
+
+    component_graphs = {
+        "Pipeline1": {
+            "Imputer": [Imputer, "X", "y"],
+            "Log Transformer": [LogTransformer, "X", "y"],
+            "RF": [RandomForestRegressor, "Imputer.x", "Log Transformer.y"],
+        },
+        "Pipeline2": {
+            "Imputer": [Imputer, "X", "y"],
+            "Log Transformer": [LogTransformer, "X", "y"],
+            "EN": [ElasticNetRegressor, "Imputer.x", "Log Transformer.y"],
+        },
+    }
+    automl = AutoMLSearch(
+        X,
+        y,
+        "regression",
+        allowed_component_graphs=component_graphs,
+        ensembling=True,
+        max_batches=4,
+        verbose=True,
+    )
+    automl.search()
+    assert "Stacked Ensemble Regression Pipeline" in caplog.text
```

Comment:
Food for thought: do you think we need to run automl search to test for this? Is there a way to test without running search / training all the other pipelines?

Reply:
This test was moreso to make sure that we can we can set pipelines in `allowed_component_graphs` and that it will create ensemble pipelines. Testing the case where we need to ensemble string/class based pipelines was more secondary. I can split off the second case into it's own test if you think that makes sense.

Reply:
Yeah, that might help give this more clarity :)